src/main/java/qna/domain/Answer.java

```diff
@@ -70,7 +73,27 @@ public boolean isOwner(User writer) {
     }
 
     public void toQuestion(Question question) {
+        if (this.question != null) {
+            this.question.getAnswers().remove(this);
+        }
         this.question = question;
+        this.question.getAnswers().add(this);
+    }
+
+    public void toWriter(User writer) {
+        if (this.writer != null) {
+            this.writer.getAnswers().remove(this);
+        }
+        this.writer = writer;
+        this.writer.getAnswers().add(this);
+    }
+
+    public DeleteHistory deleteByOwner(User owner) throws CannotDeleteException {
+        if (!this.isOwner(owner)) {
+            throw new CannotDeleteException("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
+        }
+        this.setDeleted(true);
+        return new DeleteHistory(ContentType.ANSWER, this.getId(), this.getWriter(), LocalDateTime.now());
```

Comment:
개인적인 의견인데 `DeleteHistory`에  `factory method가` 제공된다면 
좀 더 가독성 및 재 사용성이 개선될 것 같습니다 😄 

```suggestion
        return DeleteHistory.answer(this.getId(), this.getWriter());
```
이 부분은 참고만 부탁 드립니다 😄